src/gui/torrentcontentwidget.cpp

```diff
@@ -231,7 +244,7 @@ void TorrentContentWidget::keyPressEvent(QKeyEvent *event)
                                  ? Qt::Unchecked : Qt::Checked;
     const QModelIndexList selection = selectionModel()->selectedRows(TorrentContentModelItem::COL_NAME);
 
-    for (const QModelIndex &index : selection)
+    for (const QPersistentModelIndex &index : toPersistentIndexes(selection))
         model()->setData(index, state, Qt::CheckStateRole);
```

Comment:
Why not?
```suggestion
    for (const QModelIndex&index : selection)
        model()->setData(QPersistentModelIndex(index), state, Qt::CheckStateRole);
```
so you don't need to build a list beforehand.

Reply:
> so you don't need to build a list beforehand.

Because "build a list beforehand" is the whole point. Otherwise changing data of some item can invalidate other indexes.

Reply:
BTW, your suggested code does effectively nothing. It just converts QModelIndex to QPersistentModelIndex and back without any use.

Reply:
>Because "build a list beforehand" is the whole point. 

I see. Maybe consider changing to: `const QList<QPersistentModelIndex> selection = toPersistentIndexes(selectionModel()->selectedRows(TorrentContentModelItem::COL_NAME));` since the original list isn't interesting anymore.

Reply:
> Maybe consider changing to: `const QList<QPersistentModelIndex> selection = toPersistentIndexes(selectionModel()->selectedRows(TorrentContentModelItem::COL_NAME));` since the original list isn't interesting anymore.

Done.